plutus-contract/src/Language/Plutus/Contract/Tx.hs

```diff
@@ -117,17 +122,19 @@ toLedgerTx utx =
             , L.txFee = 0
             , L.txValidRange = _validityRange utx
             , L.txSignatures = Map.empty
+            , L.txData = Map.fromList $ fmap (\ds -> (L.dataScriptHash ds, ds)) (_dataValues utx)
             }
      in tx { L.txFee = minFee tx }
 
 -- | Make an unbalanced transaction that does not forge any value. Note that duplicate inputs
 --   will be ignored.
+--   Note: this doesn't populate the data scripts, so is unsafe to use directly. Prefer using 'payToScript' etc.
```

Comment:
I'd say that whatever is unsafe should either live in an `Internal` module or should contain the `unsafe` word (or an equilvalent) as a part of its name.

Reply:
It's not exported any more, this is more of a note about *why* it's no longer exported, I'll clarify.